### DISCUSSION TITLE
Merging Schemas

### DISCUSSION THREAD

**Discussion by rawkode** [cuelang/cue#872] [Q&A] [answered]
I have a Go application that uses Cue to validate the users YAML files. I store the Cue in Go as a hard coded string.

However, I want to allow the user to provide extensions to the schema, which I am loading into a Go variable as well.

The code looks like:

```go
        var model = make(map[string]interface{})

        // "cue" is a Cue string 
	err := cuego.Constrain(&model, cue)
	if err != nil {
		return nil, cueutils.UsefulError(err)
	}

	bytes, err := ioutil.ReadFile(path)
	if err != nil {
		return nil, cueutils.UsefulError(err)
	}

	err = yaml.Unmarshal(bytes, &model)
	if err != nil {
		return nil, cueutils.UsefulError(err)
	}

	err = cuego.Complete(&model)
	if err != nil {
		return nil, cueutils.UsefulError(err)
	}
```

I'd like to be able to "merge" the `cue` variable with another Go string.

Examples of `cue` and `extends`:

```go
cue := `{
  first_name: string
}`

extends := `{
  last_name: string
}`
```

I was really hoping `cuego.Merge(cue, extends)` would kind of work :)

**Comment by mpvl** [comment, marked as answer]
Note entirely sure what your flow is, but could you not just call `Constrain` repeatedly for the same value?

In general CUE, not using `cuego`, one could use `FillPath` or `Unify` for this purpose:

```
// Filling multiple Go values:

var r cue.Runtime
inst, err := r.Compile("foo", "_")
// ... handle error
v := inst.Value()

root := cue.Path{}

// Merge in Go values
v = v.FillPath(root, value1)
v = v.FillPath(root, value2)

// Merge in CUE values
inst, err = r.Compile("", cueString1)
v = v.Unify(inst.Value()) // can also use FillPath(root, v).

if err := v.Err(); err != nil {
    // handle error
}
```

**Reply by rawkode** [reply]
I did try calling Constraint twice, but it didn't function as expected.

I'll give your example code a go 👍🏼 

Thanks

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/872.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
